src/battle_script_commands.c

```diff
@@ -6667,6 +6665,23 @@ static void HandleTerrainMove(u32 moveEffect)
     }
 }
 
+bool32 DoesAbilityBypassTypeImmunity(u8 battlerAttacker, u8 battlerTarget, u8 status)
+{
+    switch(status)
+    {
+        case STATUS1_POISON:
+        case STATUS1_TOXIC_POISON:
+            if (GetBattlerAbility(battlerAttacker) == ABILITY_CORROSION)
+                return TRUE;
+            else if (IS_BATTLER_OF_TYPE(battlerTarget, TYPE_POISON))
```

Comment:
Could be simplified to
return (GetBattlerAbility(battlerAttacker) == ABILITY_CORROSION
|| !(IS_BATTLER_OF_TYPE(battlerTarget, TYPE_POISON) && IS_BATTLER_OF_TYPE(battlerTarget, TYPE_STEEL)))

Reply:
Done, but changed the final && to ||.